ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java

```diff
@@ -1026,6 +1026,31 @@ public void testFindNextCompactThrowsTException() throws Exception {
     verify(msc, times(0)).markFailed(any());
   }
 
+  @Test
+  public void testDoesNotGatherStatsIfCompactionFails() throws Exception {
+    StatsUpdater statsUpdater = Mockito.spy(StatsUpdater.class);
```

Comment:
no need for spy here, simple mock is what we need

Reply:
Modified.